apache/metron
# METRON-1507 Added support for vagrant cachier

## Contributor Comments
This is a Vagrant config change to enable the use of vagrant-cachier to reduce build times for the full dev platform. Time saving is heavily network dependent, but anecdotally I've found it saves me around 10 mins per provision of vagrant development environment. YMMV.

To test:

1. Install vagrant-cachier, and spin up centos and ubuntu vagrant development platforms.
2. Verify that packages are populated into your local (the client running vagrant) ~/.vagrant/cache directories.
3. vagrant destroy your development environments
4. spin up vagrant environments again, packages should be pulled from the cache instead of the network at this point.

Note that the vagrant-cachier plugin is not required, nor shipped, bundled or in any way distributed by this change. It does have a compatible license (MIT) but it is outside of the Metron project scope to install this, so there is no need to update license or dependency documentation.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
